### PR TITLE
Add interruptible animation feature to the Histogram example

### PR DESCRIPTION
**Pull request by aquagray**
Also adding NonAnimatingHistogramWidget so it can be linked to the medium article.

To see the final result in action see:
https://drive.google.com/a/google.com/file/d/11VLWRDBKzLPdsC1w1tCw9ZShOKuDGRyj/view?usp=sharing

### CODE REVIEW

**Comment by aquagray** [comment]
@thagikura could you take a look to see if there's anything alarming? :) If not I'll submit to get the article progressed. Thank you so much!

File: ConstraintLayoutExamples/motionlayout/src/main/java/com/google/androidstudio/motionlayoutexample/histogramdemo/HistogramAnimationGuard.kt [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        override fun onTransitionTrigger(p0: MotionLayout?, p1: Int, p2: Boolean, p3: Float) { }
+        override fun onTransitionChange(p0: MotionLayout?, p1: Int, p2: Int, p3: Float) { }
+    }
+
+    /**
+     * @param interruptible true if animation is interruptible. I.e. Animation does not need to be
+     * finished before the new one starts.
+     * False if animation must complete before new one can start.
+     */
+    fun setInterruptible(interruptible: Boolean) {
```
Comment (thagikura) [inline ConstraintLayoutExamples/motionlayout/src/main/java/com/google/androidstudio/motionlayoutexample/histogramdemo/HistogramAnimationGuard.kt]: Kotlin automatically generates a pair of getter/setter for a member. 
I think you don't have to create a setter for it.
If you want to change the visibility of the getter or setter, you can write like

var interruptible: Boolean = false
    private set

Please see https://kotlinlang.org/docs/reference/properties.html

Comment (thagikura) [reply ConstraintLayoutExamples/motionlayout/src/main/java/com/google/androidstudio/motionlayoutexample/histogramdemo/HistogramAnimationGuard.kt]: I think this is not a blocker. Let's merge the PR as of now and I appreciate if you can consider fixing it in a follow up PR if you agree.

Comment (aquagray) [reply ConstraintLayoutExamples/motionlayout/src/main/java/com/google/androidstudio/motionlayoutexample/histogramdemo/HistogramAnimationGuard.kt]: Awesome :) Thanks! 

I've created:
https://github.com/android/views-widgets-samples/pull/104

as a followup

**Pull request merged**